precompile/contracts/deployerallowlist/contract_test.go

```diff
@@ -4,202 +4,13 @@
-			ReadOnly:    true,
-			ExpectedErr: vmerrs.ErrWriteProtection.Error(),
-		},
-		"set no role insufficient gas": {
-			Caller: adminAddr,
-			Input: func(tt *testing.T) []byte {
-				input, err := allowlist.PackModifyAllowList(adminAddr, allowlist.EnabledRole)
-				require.NoError(tt, err)
-
-				return input
-			}(t),
-			SuppliedGas: allowlist.ModifyAllowListGasCost - 1,
-			ReadOnly:    false,
-			ExpectedErr: vmerrs.ErrOutOfGas.Error(),
-		},
-		"read allow list no role": {
-			Caller:      noRoleAddr,
-			Input:       allowlist.PackReadAllowList(enabledAddr),
-			SuppliedGas: allowlist.ReadAllowListGasCost,
-			ReadOnly:    false,
-			ExpectedRes: common.Hash(allowlist.EnabledRole).Bytes(),
-		},
-		"read allow list admin role": {
-			Caller:      adminAddr,
-			Input:       allowlist.PackReadAllowList(enabledAddr),
-			SuppliedGas: allowlist.ReadAllowListGasCost,
-			ReadOnly:    false,
-			ExpectedRes: common.Hash(allowlist.EnabledRole).Bytes(),
-		},
-		"read allow list with readOnly enabled": {
-			Caller:      adminAddr,
-			Input:       allowlist.PackReadAllowList(enabledAddr),
-			SuppliedGas: allowlist.ReadAllowListGasCost,
-			ReadOnly:    true,
-			ExpectedRes: common.Hash(allowlist.EnabledRole).Bytes(),
-		},
-		"read allow list out of gas": {
-			Caller:      adminAddr,
-			Input:       allowlist.PackReadAllowList(noRoleAddr),
-			SuppliedGas: allowlist.ReadAllowListGasCost - 1,
-			ReadOnly:    true,
-			ExpectedErr: vmerrs.ErrOutOfGas.Error(),
-		},
-	} {
-		t.Run(name, func(t *testing.T) {
-			db := rawdb.NewMemoryDatabase()
-			state, err := state.New(common.Hash{}, state.NewDatabase(db), nil)
-			require.NoError(t, err)
-
-			// Set up the state so that each address has the expected permissions at the start.
-			SetContractDeployerAllowListStatus(state, adminAddr, allowlist.AdminRole)
-			SetContractDeployerAllowListStatus(state, enabledAddr, allowlist.EnabledRole)
-			SetContractDeployerAllowListStatus(state, noRoleAddr, allowlist.NoRole)
-			require.Equal(t, allowlist.AdminRole, GetContractDeployerAllowListStatus(state, adminAddr))
-			require.Equal(t, allowlist.EnabledRole, GetContractDeployerAllowListStatus(state, enabledAddr))
-			require.Equal(t, allowlist.NoRole, GetContractDeployerAllowListStatus(state, noRoleAddr))
-
-			blockContext := contract.NewMockBlockContext(common.Big0, 0)
-			accesibleState := contract.NewMockAccessibleState(state, blockContext, snow.DefaultContextTest())
-			ret, remainingGas, err := ContractDeployerAllowListPrecompile.Run(accesibleState, test.Caller, ContractAddress, test.Input, test.SuppliedGas, test.ReadOnly)
-			if len(test.ExpectedErr) != 0 {
-				require.ErrorContains(t, err, test.ExpectedErr)
-			} else {
-				require.NoError(t, err)
-			}
-
-			require.Equal(t, uint64(0), remainingGas)
-			require.Equal(t, test.ExpectedRes, ret)
-
-			if test.AfterHook != nil {
-				test.AfterHook(t, state)
-			}
-		})
-	}
+	// TODO: add module specific tests
```

Comment:
Can we remove this TODO since there are no module specific tests for the deployer allow list?